app/lib/actions/katello/repository/destroy_medium.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Katello
+    module Repository
+      class DestroyMedia < Actions::Base
+        def plan(repo)
+          plan_self(:repo_id => repo.id)
+        end
+
+        def finalize
+          repo = ::Katello::Repository.find(input[:repo_id])
+          medium = Medium.find_medium(repo)
+          medium.destroy if medium
```

Comment:
I would create an implement a destroy! method on medium and also use this in the plan phase so that if it fails it fails up front.  My worry is this causes a problem: https://github.com/theforeman/foreman/blob/develop/app/models/medium.rb#L12